src/actions/breakpoints.js

```diff
@@ -407,6 +408,52 @@ export function toggleBreakpoint(line: ?number, column?: number) {
   };
 }
 
+export function toggleBreakpointsAtLine(line: number, column?: number) {
+  return ({ dispatch, getState, client, sourceMaps }: ThunkArgs) => {
+    const state = getState();
+    const selectedSource = getSelectedSource(state);
+
+    if (!line || !selectedSource) {
+      return;
+    }
+
+    const bps = getBreakpointsAtLine(state, line);
+    const isEmptyLine = isEmptyLineInSource(state, line, selectedSource.id);
+
+    if (isEmptyLine) {
+      return;
+    }
+
+    if (bps.size === 0) {
+      return dispatch(
+        addBreakpoint({
+          sourceId: selectedSource.id,
+          sourceUrl: selectedSource.url,
+          line: line,
+          column: column
+        })
+      );
+    }
+
+    const promises = bps.map(bp => {
+      if (bp.loading) {
+        return null;
+      }
+
+      return dispatch(
+        removeBreakpoint({
+          sourceId: bp.location.sourceId,
+          sourceUrl: bp.location.sourceUrl,
+          line: bp.location.line,
+          column: column || bp.location.column
+        })
+      );
+    });
+
+    return Promise.all(promises);
```

Comment:
I wonder if we can do something like this:

```js
return Promise.all(bps.map(bp => dispatch(removeBreakpoint(bp.location))
```

the advantage is that we are not duplicating logic

Reply:
Yeah, we can tighten that up.